tools/blast_parser/blast_parser.py

```diff
@@ -35,21 +52,43 @@ def main():
 
         # Ignore self-matching hits
         if sequence1_id != sequence2_id:
+            i = i + 1
+
             # Convert evalue to an integer weight with max 100
             weight = 100
 
             # If the evalue is 0, leave weight at 100
             if evalue != 0.0:
                 weight = min(100, round(math.log10(evalue) / -2.0))
 
-            if (sequence1_id, sequence2_id) not in results:
-                results[(sequence1_id, sequence2_id)] = weight
-            else:
-                results[(sequence1_id, sequence2_id)] = max(results[(sequence1_id, sequence2_id)], weight)
+            # Insert pair into SQLite database
+            cur.execute('INSERT INTO homology (homology_id, sequence1_id, sequence2_id, weight) VALUES (?, ?, ?, ?)',
+                        (i, sequence1_id, sequence2_id, weight))
+
+            # Execute query at every 100 pair to save memory
+            if(i % 100 == 0):
+                conn.commit()
+
+    # Execute query at last
+    conn.commit()
+
+    # Delete duplicate keeping one with max weight
+    cur.execute('delete FROM homology WHERE homology_id  IN (SELECT a1.homology_id FROM homology a1 INNER JOIN homology a2 ON a1.sequence1_id = a2.sequence1_id AND a1.sequence2_id = a2.sequence2_id AND a1.weight < a2.weight)')
```

Comment:
The following query should be much faster and require less memory (no join needed):
```sql
DELETE FROM alignment WHERE id NOT IN (SELECT id FROM alignment GROUP BY sequence1_id, sequence2_id HAVING max(weight));
```